src/services/reports.js

```diff
@@ -0,0 +1,359 @@
+						const additionalFilter =
+							filterType == common.MENTEE_ROLE
+								? {
+										mentee_visibility_policy: {
+											//1
+											[Op.ne]: 'CURRENT',
+										},
+								  }
+								: filterType == common.SESSION
+								? {
+										session_visibility_policy: {
+											//2
+											[Op.ne]: 'CURRENT',
+										},
+								  }
+								: {
+										mentor_visibility_policy: {
+											//3
+											[Op.ne]: 'CURRENT',
+										},
+								  }
+						const organizationExtension = await organisationExtensionQueries.findAll(
+							{
+								[Op.or]: [
+									filterQuery,
+									{
+										[Op.and]: [
+											{
+												organization_id: {
+													[Op.in]: [...relatedOrgs],
+												},
+											},
+											additionalFilter,
+										],
+									},
+								],
+							},
+							{
+								attributes: ['organization_id'],
+							}
+						)
+						organizationIds.push(orgExtension.organization_id)
+						if (organizationExtension) {
+							const organizationIdsFromOrgExtension = organizationExtension.map(
+								(orgExt) => orgExt.organization_id
+							)
+							organizationIds.push(...organizationIdsFromOrgExtension)
+						}
+					}
+				}
+			}
+
+			return {
+				success: true,
+				result: organizationIds,
+			}
+		} catch (error) {
+			return {
+				success: false,
+				message: error.message,
+			}
+		}
+	}
+
+	static async getEntityTypeWithEntitiesBasedOnOrg(organization_ids, entity_types, defaultOrgId = '', modelName) {
+		try {
+			let filter = {
+				status: common.ACTIVE_STATUS,
+				allow_filtering: true,
+				has_entities: true,
+				organization_id: {
+					[Op.in]: defaultOrgId ? [...organization_ids, defaultOrgId] : organization_ids,
+				},
+				report_filter: true,
+			}
+
+			let entityTypes = []
+			if (entity_types) {
+				entityTypes = entity_types.split(',')
+				filter.value = {
+					[Op.in]: entityTypes,
+				}
+			}
+
+			if (modelName) {
+				filter.model_names = { [Op.contains]: [modelName] }
+			}
+			//fetch entity types and entities
+			let entityTypesWithEntities = await entityTypeQueries.findUserEntityTypesAndEntities(filter)
+
+			return {
+				success: true,
+				result: entityTypesWithEntities,
+			}
+		} catch (error) {
+			return {
+				success: false,
+				message: error.message,
+			}
+		}
+	}
+}
+
+function convertEntitiesForFilter(entityTypes) {
+	const result = {}
+
+	entityTypes.forEach((entityType) => {
+		const key = entityType.value
+
+		if (!result[key]) {
+			result[key] = []
+		}
+
+		const newObj = {
+			id: entityType.id,
+			label: entityType.label,
+			value: entityType.value,
+			parent_id: entityType.parent_id,
+			organization_id: entityType.organization_id,
+			entities: entityType.entities || [],
+		}
+
+		result[key].push(newObj)
+	})
+	return result
+}
+
+function filterEntitiesBasedOnParent(data, defaultOrgId, doNotRemoveDefaultOrg) {
```

Comment:
@sumanvpacewisdom don't duplicate the code in multiple places, move it one file use it in both relevant functions